buildSrc/src/main/groovy/com/uber/okbuck/core/dependency/DependencyCache.groovy

```diff
@@ -1,293 +1,260 @@
-
-            // Copy the file into the cache
-            if (!cachedCopy.exists()) {
-                Files.createSymbolicLink(cachedCopy.toPath(), e.depFile.toPath())
-            }
-            cachedCopies.add(cachedCopy)
-
-            // Extract Lint Jars
-            if (extractLintJars && cachedCopy.name.endsWith(".aar")) {
-                File lintJar = getPackagedLintJarFrom(cachedCopy)
-                if (lintJar != null) {
-                    String lintJarPath = FileUtil.getRelativePath(rootProject.projectDir, lintJar)
-                    lintJars.put(e, lintJarPath)
-                    cachedCopies.add(lintJar)
-                }
-            }
-
-            // Fetch Sources
-            if (fetchSources) {
-                File sources = fetchSourcesFor(e)
-                if (sources != null) {
-                    cachedCopies.add(sources)
+    /**
+     * Get the list of annotation processor classes provided by a dependency.
+     *
+     * @param dependency The dependency
+     * @return The list of annotation processor classes available in the manifest
+     */
+    List<String> getAnnotationProcessors(ExternalDependency dependency) {
+        String key = dependency.cacheName
+        String processorsList = processors.get(key)
+        if (processorsList == null) {
+            JarFile jarFile = new JarFile(dependency.depFile)
+            JarEntry jarEntry = (JarEntry) jarFile.getEntry("META-INF/services/javax.annotation.processing.Processor")
+            if (jarEntry) {
+                List<String> processorClasses = IOUtils.toString(jarFile.getInputStream(jarEntry))
+                        .trim().split("\\n").findAll { String entry ->
+                    !entry.startsWith('#') && !entry.trim().empty // filter out comments and empty lines
                 }
+                processorsList = processorClasses.join(",")
+            } else {
+                processorsList = ""
             }
+            processors.set(key, processorsList)
         }
 
-        // cleanup
-        if (cleanup) {
-            (cacheDir.listFiles(new FileFilter() {
-
-                @Override
-                boolean accept(File pathname) {
-                    return pathname.isFile() && (pathname.name.endsWith(".jar") || pathname.name.endsWith(".aar"))
-                }
-            }) - cachedCopies).each { File f ->
-                Files.deleteIfExists(f.toPath())
-            }
+        if (processorsList == "") {
+            return Collections.emptyList()
+        } else {
+            return processorsList.split(",")
         }
     }
 
-    static boolean isWhiteListed(File depFile) {
-        return WHITELIST_LOCAL_PATTERNS.find { depFile.absolutePath.contains(it) } != null
+    /**
+     * Get the packaged lint jar of an aar dependency if any.
+     *
+     * @param dependency The depenency
+     * @return path to the lint jar in the cache.
+     */
+    String getLintJar(ExternalDependency dependency) {
```

Comment:
lint is part of it because dependencies can have embedded metadata like lint jars. The rest of the lint logic is already in a separate class